Automattic/jetpack
# Fixes #5788 by making sure that do_action( 'wp_insert_post'

is being called with all arguments. Otherwise bail.

Some plugins namely https://wordpress.org/plugins/pet-manager/ call
do_action in a strange way. This PR fixes any warning that would be
displayed because of the the values are not passed in as expected.

Fixes #5788 

#### Changes proposed in this Pull Request:
- Adds checks to make sure that the data passed into do_action is what we expect.

#### Testing instructions:
* Do the tests pass? 
* install the plugin https://wordpress.org/plugins/pet-manager/
and notice that the php warnings are not there any more.
